broker/unified_sql/src/stream_sql.cc

```diff
@@ -5169,6 +5181,50 @@ void stream::_process_tag(const std::shared_ptr<io::data>& d) {
   }
 }
 
+void stream::_process_agent_stats(const std::shared_ptr<io::data>& d) {
+  SPDLOG_LOGGER_INFO(_logger_sql, "unified_sql: processing agent stats");
+  std::shared_ptr<neb::pb_agent_stats> as{
+      std::static_pointer_cast<neb::pb_agent_stats>(d)};
+
+  std::string json_infos;
+
+  const AgentStats& stats = as->obj();
+
+  using namespace rapidjson;
+  Document doc(rapidjson::kArrayType);
+
+  for (const AgentInfo& info : stats.stats()) {
+    rapidjson::Value stat(rapidjson::kObjectType);
+    stat.AddMember("agent_major", info.major(), doc.GetAllocator());
+    stat.AddMember("agent_minor", info.minor(), doc.GetAllocator());
+    stat.AddMember("agent_patch", info.patch(), doc.GetAllocator());
+    stat.AddMember("reverse", info.reverse(), doc.GetAllocator());
+    stat.AddMember("os", StringRef(info.os().c_str()), doc.GetAllocator());
+    stat.AddMember("os_version", StringRef(info.os_version().c_str()),
+                   doc.GetAllocator());
+    stat.AddMember("nb_agent", info.nb_agent(), doc.GetAllocator());
+    doc.PushBack(stat, doc.GetAllocator());
+  }
+  StringBuffer out_buff;
+  Writer<StringBuffer> writer(out_buff);
+  doc.Accept(writer);
```

Comment:
Seems simpler to directly write the string with the json with a good fmt::format...